### PR TITLE
Fix error in `TwoFactorPamAuthenticatable` when handling invalid params

### PR DESCRIPTION
**Pull request by ClearlyClaire**
This is a very minor edge case, but it shows up in tests in some forks now that we are more extensively testing invalid parameters.

### CODE REVIEW

File: lib/devise/strategies/two_factor_pam_authenticatable.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def authenticate!
       protected
 
       def valid_params?
-        params[scope] && params[scope][:password].present?
+        params[scope].respond_to?(:[]) && params[scope][:password].present?
```
Comment (oneiros) [inline lib/devise/strategies/two_factor_pam_authenticatable.rb:25]: Would

```ruby
params.dig(scope, :password).present?
```

work as well? That is a bit more succinct and imo even a bit more readable.

Comment (ClearlyClaire) [reply lib/devise/strategies/two_factor_pam_authenticatable.rb:25]: No! This would precisely not work, because the failing test passes an array as `scope`, which causes `no implicit conversion of Symbol into Integer`, an error that would remain the same with `dig`!

Comment (oneiros) [reply lib/devise/strategies/two_factor_pam_authenticatable.rb:25]: Ah, OK. Thanks for clarifying!

**Review by oneiros — APPROVED** [review]
